File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {
        int left = 0;
        int ans = 0;

        int maxf = 0;
        int[] record = new int[128];
    
        for (int right = 0 ; right < s.length(); right++) {
            char r = s.charAt(right);
            record[r]++;
            maxf = Math.max(maxf, record[r]);

            while ((right - left + 1) - maxf > k) {
                record[s.charAt(left)]--;
                left++;
            }
            ans = Math.max(ans, right - left + 1);
        }
        return ans;

    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        charSet = set()
        ans = 0

        for c in list(s):
            while c in charSet:
                charSet.remove(s[left])
                left += 1
            charSet.add(c)
            ans = max(ans, len(charSet))

        return ans

        

        